File: app/src/main/java/tpcreative/co/qrscanner/common/preference/MyPreferenceCategory.java
package tpcreative.co.qrscanner.common.preference;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.TextView;
import androidx.preference.PreferenceCategory;
import androidx.preference.PreferenceViewHolder;
import tpcreative.co.qrscanner.R;


public class MyPreferenceCategory extends PreferenceCategory {

    public MyPreferenceCategory(Context context) {
        super(context);
    }

    public MyPreferenceCategory(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public MyPreferenceCategory(Context context, AttributeSet attrs,int defStyle) {
        super(context, attrs, defStyle);
    }

    @Override
    public void onBindViewHolder(PreferenceViewHolder view) {
        super.onBindViewHolder(view);
        TextView titleView = (TextView) view.findViewById(android.R.id.title);
        titleView.setTextColor(getContext().getResources().getColor(R.color.colorPrimary));
        titleView.setAllCaps(true);
        titleView.setTextSize(17);
    }

}

File: app/src/main/java/tpcreative/co/qrscanner/common/preference/MyPreferenceCategory.kt
package tpcreative.co.qrscanner.common.preference

import android.content.*
import android.util.AttributeSet
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.preference.PreferenceCategory
import androidx.preference.PreferenceViewHolder
import tpcreative.co.qrscanner.R
import tpcreative.co.qrscanner.common.services.QRScannerApplication

class MyPreferenceCategory @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet,
        defStyleAttr: Int = 0
) : PreferenceCategory(context, attrs, defStyleAttr) {
    init {
        widgetLayoutResource = R.layout.custom_preferences_categories
    }

    override fun onBindViewHolder(view: PreferenceViewHolder?) {
        super.onBindViewHolder(view)
        val titleView = view?.findViewById(android.R.id.title) as TextView
        titleView.setTextColor(ContextCompat.getColor(QRScannerApplication.getInstance(), R.color.colorPrimary))
        titleView.isAllCaps = true
        titleView.textSize = 17f
    }
}